File: AspNetMvcUnleashed/0672329980_source/Chapter07/VB/MvcApplication1/MvcApplication1/Controllers/Movie2Controller.vb
Public Class Movie2Controller
    Inherits System.Web.Mvc.Controller

    ' GET: /Movie2/Create
    Function Create() As ActionResult
        Return View()
    End Function

    ' POST: /Movie2/Create
    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(ByVal collection As FormCollection) As ActionResult
        Dim movieToCreate As New Movie()
        Me.UpdateModel(movieToCreate, collection.ToValueProvider())
        ' Insert movie into database
        Return RedirectToAction("Index")
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter07/CS/MvcApplication1/MvcApplication1/Controllers/Movie2Controller.cs
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class Movie2Controller : Controller
    {
        // GET: /Movie2/Create
        public ActionResult Create()
        {
            return View();
        } 

        // POST: /Movie2/Create
        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(FormCollection collection)
        {
            var movieToCreate = new Movie();
            this.UpdateModel(movieToCreate, collection.ToValueProvider());
            // Insert movie into database
            return RedirectToAction("Index");
        }
    }
}
